File: tests/output/csharp/src/Models/TopModel.Sample.Restaurant.Models/PlatItem.cs
////
//// ATTENTION CE FICHIER EST GENERE AUTOMATIQUEMENT !
////

namespace TopModel.Sample.Restaurant.Models;

public class PlatItem : IPlatItem
{
    public int? Id => throw new NotImplementedException();

    public string? Nom => throw new NotImplementedException();

    public decimal? Prix => throw new NotImplementedException();

    public bool? Disponible => throw new NotImplementedException();

    public CategoriePlat.Codes? CategoriePlatCode => throw new NotImplementedException();

    public static IPlatItem Create(
        int? id = null,
        string? nom = null,
        decimal? prix = null,
        bool? disponible = null,
        CategoriePlat.Codes? categoriePlatCode = null
    )
    {
        throw new NotImplementedException();
    }
}

File: tests/output/jpa/src/main/javagen/restaurant/jpa_identity_feign/dtos/restaurant/PlatItem.java
////
//// ATTENTION CE FICHIER EST GENERE AUTOMATIQUEMENT !
////

package restaurant.jpa_identity_feign.dtos.restaurant;

import java.math.BigDecimal;

import jakarta.annotation.Generated;

import restaurant.jpa_identity_feign.enums.restaurant.CategoriePlat;

/**
 * Détail d'un plat en liste.
 */
@Generated("TopModel : https://github.com/klee-contrib/topmodel")
public interface PlatItem {

	/**
	 * Identifiant du plat.
	 */
	Integer getId();

	/**
	 * Nom du plat.
	 */
	String getNom();

	/**
	 * Prix du plat.
	 */
	BigDecimal getPrix();

	/**
	 * Indique si le plat est disponible.
	 */
	Boolean getDisponible();

	/**
	 * Catégorie du plat.
	 */
	CategoriePlat getCategoriePlatCode();

	/**
	 * Hydrate values of instance.
	 * @param id value to set.
	 * @param nom value to set.
	 * @param prix value to set.
	 * @param disponible value to set.
	 * @param categoriePlatCode value to set.
	 */
	void hydrate(Integer id, String nom, BigDecimal prix, Boolean disponible, CategoriePlat categoriePlatCode);
}
